crates/flox-rust-sdk/src/providers/git.rs

```diff
@@ -283,6 +283,62 @@ impl GitCommandProvider {
 
         Ok(OsString::from_vec(out.stdout))
     }
+
+    /// open repo, erroring if path is not a repo or is a subdirectory of a repo
+    pub fn open<P: AsRef<Path>>(path: P) -> Result<Self, GitCommandOpenError> {
+        let path = path.as_ref().to_path_buf();
+
+        let out = GitCommandProvider::run_command(
+            GitCommandProvider::new_command(&Some(&path))
+                .arg("rev-parse")
+                .arg("--is-bare-repository"),
+        )?;
+
+        let out_str = out
+            .to_str()
+            .ok_or(GitCommandDiscoverError::GitDirEncoding)?;
+
+        let bare = out_str
```

Comment:
Suggestion: if you don't need `out_str` except to extract `bare` you can probably chain these together.

Reply:
I copied that from `discover()` - want me to change both in this PR? I don't think we need `out` either - would you just chain it all the way?

Reply:
It's my personal preference to chain the whole thing, but the choice is ultimately up to you. If you change it here I would also change it in `discover`.

Reply:
i think there were issues with the borrow checker preventing chaining this further iirc.
If that's not the case anymore, or you solve it in a different way, i agree with zach we should do it in both methdos